### PR TITLE
Report datadog.tracer.abandoned_spans health metric

### PR DESCRIPTION
**Pull request by mtoffl01**
### What does this PR do?
Report the count of unfinished spans as a metric, along with tags about the span name and span integration.

This PR also makes the following small changes:
- Amends abandonedSpansCandidate type to contain an Integration field
- Modifies the TestStatsdClient for ease of use

### Motivation
Abandoned spans info is already available in tracer logs, but making the information visible as a metric that we can track alongside other data points in the Datadog platform will make debugging easier for customer and support.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.
- [ ] For internal contributors, a matching PR should be created to the `v2-dev` branch and reviewed by @DataDog/apm-go.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: ddtrace/tracer/abandonedspans_test.go [hunk 1]
```diff
@@ -58,11 +60,62 @@ func assertProcessedSpans(assert *assert.Assertions, t *tracer, startedSpans, fi
 
 func formatSpanString(s *span) string {
 	s.Lock()
-	msg := fmt.Sprintf("[name: %s, span_id: %d, trace_id: %d, age: %s],", s.Name, s.SpanID, s.TraceID, spanAge(s))
+	var integration string
+	if v, ok := s.Meta[ext.Component]; ok {
+		integration = v
+	} else {
+		integration = "manual"
```
Comment (darccio) [inline ddtrace/tracer/abandonedspans_test.go:67]: @mtoffl01 @hannahkm QQ: wouldn't a blank `integration` already indicate it's a span created manually? Is this required somewhere else to have `manual` as value?

Comment (hannahkm) [reply ddtrace/tracer/abandonedspans_test.go:67]: I believe the main concern is that leaving it blank might give the impression of being misconfigured/missing information. If we _manually_ set it to manual here, it'll give a little more clarity.

Comment (mtoffl01) [reply ddtrace/tracer/abandonedspans_test.go:67]: Yes, exactly what Hannah said.
From being on the troubleshooting side, an explicit value (e.g. "manual") is a good way to differentiate from an unset/missing value.

Comment (darccio) [reply ddtrace/tracer/abandonedspans_test.go:67]: LGTM then

**Pull request merged**